src/inter_dc_communication_sender.erl

```diff
@@ -132,7 +125,7 @@ stop(timeout, State=#state{socket=Socket}) ->
 
 stop_error(timeout, State=#state{socket=Socket}) ->
     _ = gen_tcp:close(Socket),
-    {stop, error, State}.
+    {stop, normal, State}.
```

Comment:
Yes because there is code to restart it in the propagate sync function. So before there were two places checking for timeouts.

Actually this is a question I have for the code in general. Is there a general rule for when is best for using the supervisor to  restart and when to do it manually. Bad messages? Connection timeouts? Receive timeouts? Etc.. Or is it just decided depending on the situation?